ckanext/switzerland/logic.py

```diff
@@ -430,10 +428,9 @@ def ogdch_user_create(context, data_dict):
     if send_email_on_registration and user.get('email'):
         try:
             send_registration_email(user)
-        except (socket_error, mailer.MailerException) as error:
-            h.flash_warning("The email could not be send to {} for user {}. An error {} occured"  # noqa
-                            .format(user['name'], user['email'], error))  # noqa
-        else:
             h.flash_success("An email has been send to the user {} at {}."  # noqa
                             .format(user['name'], user['email']))
+        except Exception as e:
+            h.flash_error("The email could not be send to {} for user {}. An error {} occured"  # noqa
+                            .format(user['name'], user['email'], e))  # noqa
```

Comment:
Not related to your fix, but I just noticed that the format values are in the wrong order here - I think it should be `user['email'], user['name'], e`